contracts/defi/interfaces/IRewarderNew.sol

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+interface IRewarderNew {
+  function onReward(address user) external returns (uint256 index);
+  
+  function claimReward(
+    uint256 objectiveIndex,
+    uint256 interactionIndex,
+    address originAddress,
+    address user
+  ) external; 
+
+  function getObjectiveByOrigin(address origin) external view returns (int256 index); 
```

Comment:
This returns a single index? There might be more than one for an origin, I think it should be clearer which one we'll return in that case.

Maybe we don't even need this one?

Reply:
if there were many objectives for one origin, how would we get the `objectiveIndex`? Right now, I think we can only do it off-chain by listening to the events (or going through the `objectivesByOrigin` mapping). I'm thinking about the case where we would claim the reward directly from the cotract that triggered the interaction.

Reply:
I think in that case the contract that triggered the interaction would need to receive the index as a parameter to pass it to the rewarder.

I guess the other option is to use the last objetive and see if it matches. In most cases I think it will, although it's not 100% sure.